### ISSUE TITLE
Exception in Rosie query

### ISSUE THREAD

**Issue by ScottWales** [metomi/rose#1567]
The `_rows_to_maps()` function in lib/python/rosie/db.py is not checking if the variable `name` is a string before using the `endswith()` function. We are getting this error in the web interface when doing a query:

```
Traceback (most recent call last):
  File "/usr/lib/python2.6/site-packages/cherrypy/_cprequest.py", line 670, in respond
    response.body = self.handler()
  File "/usr/lib/python2.6/site-packages/cherrypy/lib/encoding.py", line 212, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/usr/lib/python2.6/site-packages/cherrypy/_cpdispatch.py", line 61, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/usr/local/rose/default/lib/python/rosie/ws.py", line 90, in query
    data = self.dao.query(filters, all_revs)
  File "/usr/local/rose/default/lib/python/rosie/db.py", line 232, in query
    return_maps = self._rows_to_maps(values_list, cols)
  File "/usr/local/rose/default/lib/python/rosie/db.py", line 397, in _rows_to_maps
    if name.endswith("-list") and value is not None:
AttributeError: 'NoneType' object has no attribute 'endswith'
```

**Comment by benfitzpatrick** [comment]
What query causes the name to be `None`?

**Comment by ScottWales** [comment]
It was an owner query for the string 'jtl548'. Might be something wrong with our database?

On 26 Mar 2015, at 7:55 pm, Ben Fitzpatrick <notifications@github.com<mailto:notifications@github.com>> wrote:


What query causes the name to be None?

**Comment by matthewrmshin** [comment]
@ScottWales There is something incorrect in your `optional` table of your database. Otherwise, I cannot see how you will end up with `name` being `None`. Do you think you can have a look at the `sqlite` file to see what is happening to that table?

**Comment by benfitzpatrick** [comment]
Maybe try rebuilding (`rosa db-create`) the database to a different file and see if it still has the strange `None` or null string row?

**Comment by ScottWales** [comment]
Doing some testing with different Rose versions it looks like this may be due to us running an old version of the webserver (2014-05) with newer client versions. I don't see the same issue when I use the latest version of the webserver. I'll put in a workaround for the moment, and hopefully we'll be able to update our site defaults soon.

**Comment by matthewrmshin** [comment]
Perhaps this was fixed by #1409?
